lib/puppet/parser/functions/hiera_hash.rb

```diff
@@ -29,7 +29,7 @@ module Puppet::Parser::Functions
   found in the data sources are strings or arrays, puppet will raise a type mismatch error.
 
   More thorough examples of `hiera_hash` are available at:  
-  <http://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
+  <https://docs.puppetlabs.com/hiera/1/puppet.html#hiera-lookup-functions>
```

Comment:
Probably best to use this link now - https://docs.puppetlabs.com/hiera/3.0/puppet.html#hiera-lookup-functions

Example will need checked though.